numismatic/cli.py

```diff
@@ -161,64 +161,27 @@ def tabulate(data):
+@click.option('--channels', '-C', multiple=True, default=['trades'])
 @click.option('--api-key-id', default=None)
 @click.option('--api-key-secret', default=None)
 @pass_state
 def listen(state, feed, assets, currencies, raw_output, raw_interval, 
-           channel, api_key_id, api_key_secret):
+           channels, api_key_id, api_key_secret):
     'Listen to live events from a feed'
     feed_name = feed
     assets = ','.join(assets).upper().split(',')
     currencies = ','.join(currencies).upper().split(',')
-    pairs = list(map('/'.join, product(assets, currencies)))
-    output_stream = state['output_stream']
-    subscriptions = state['subscriptions']
-    # FIXME: each pair should get a separate stream which we should keep track
-    #        of in state['streams'].
-    if feed_name=='bitfinex':
-        # FIXME: Move the pairs handling into the Feed code
-        for pair in pairs:
-            pair = pair.replace('/', '')
-            feed = Feed.factory(feed_name, 
-                                output_stream=output_stream,
-                                raw_stream=raw_output,
-                                raw_interval=raw_interval)
-            subscription = feed.listen(pair, channel)
-            subscriptions[f'{pair}-{feed_name}'] = subscription
-    elif feed_name=='gdax':
-        if channel=='trades':
-            # FIXME: handle this mapping in a better way
-            channel = 'ticker'
-        for pair in pairs:
-            pair = pair.replace('/', '-')
-            feed = Feed.factory(feed_name,
-                                output_stream=output_stream,
-                                raw_stream=raw_output,
-                                raw_interval=raw_interval)
-            subscription = feed.listen(pair, channel)
-            subscriptions[f'{pair}-{feed_name}'] = subscription
-    elif feed_name=='luno':
-        if api_key_id is None:
-            api_key_id = (config['Luno'].get('api_key_id', '') if 'Luno' in
-                          config else '')
-            api_key_secret = (config['Luno'].get('api_key_secret', '') if
-                              'Luno' in config else '')
-        feed = Feed.factory(feed_name,
-                            output_stream=output_stream,
-                            raw_stream=raw_output,
-                            raw_interval=raw_interval,
-                            api_key_id=api_key_id,
-                            api_key_secret=api_key_secret)
-        for pair in pairs:
-            pair = pair.replace('/', '')
-            subscription = feed.listen(pair)
-            subscriptions[f'{pair}-{feed_name}'] = subscription
-    else:
-        raise NotImplementedError()
+    channels = ','.join(channels).lower().split(',')
+    feed = Feed.factory(feed_name) 
+    subscriptions = feed.subscribe(assets, currencies, channels)
+    state['subscriptions'].update(subscriptions)
+    print(state['subscriptions'].keys())
```

Comment:
I am assuming this print statement is for debugging? If so, please remove it from code

Reply:
yeah, gone in the sql PR